src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckPolicyViolationCollector.java

```diff
@@ -137,17 +148,18 @@ private PolicyComponentMapping createPolicyComponentMapping(final ComponentVersi
         return new PolicyComponentMapping(componentName, policies);
     }
 
-    private BlackDuckPolicyLinkableItem createBlackDuckPolicyLinkableItem(final ComponentVersionStatus componentVersionStatus) {
+    private BlackDuckPolicyLinkableItem createBlackDuckPolicyLinkableItem(final ComponentVersionStatus componentVersionStatus, final String projectVersionWithComponentLink) {
         final BlackDuckPolicyLinkableItem blackDuckPolicyLinkableItem = new BlackDuckPolicyLinkableItem();
 
-        final String componentName = componentVersionStatus.getComponentName();
-        if (StringUtils.isNotBlank(componentName)) {
-            blackDuckPolicyLinkableItem.addComponentNameItem(componentName, componentVersionStatus.getComponent());
-        }
-
         final String componentVersionName = componentVersionStatus.getComponentVersionName();
         if (StringUtils.isNotBlank(componentVersionName)) {
-            blackDuckPolicyLinkableItem.addComponentVersionItem(componentVersionName, componentVersionStatus.getComponentVersion());
+            blackDuckPolicyLinkableItem.addComponentVersionItem(componentVersionName, projectVersionWithComponentLink);
+        }
+
+        final String componentName = componentVersionStatus.getComponentName();
+        if (StringUtils.isNotBlank(componentName)) {
+            final String componentLink = (StringUtils.isBlank(componentVersionName)) ? projectVersionWithComponentLink : componentVersionStatus.getComponent();
```

Comment:
I don't think this inline condition is correct. The link should always be the same. If a the component version is missing then the link should be applied to the component name, if the component version is present then the link should be applied to the version

Reply:
I can make this clearer by changing componentVersionStatus.getComponent() to null. The idea is if we have a component name, but the component version is blank, we add the link to the name. If we don't want a link we add null. The value I used instead of null HAPPENS to be null if there is no version, so I am not sure why I used that instead of just null.